readme.md
# Trapped in Space Game

## Changelog.md available in the docs above

## Github Pages Deployment
https://iesnoth.github.io/Trapped-in-Space/

## Important Info for testers
- Has to be played on Chrome
- If you get the doom screen, click back button to reset the game

### A Brief Overview
Have you ever wondered:
"What would it be like to be the pilot of a lo-fi 16-bit spaceship, but I still don't know what any of the buttons do?"
Well, this game seeks to satisfy that very specific curiosity.
There are a lot of buttons, you push a lot of buttons, and sometimes they do things.

Sometimes they do very bad things.

It's a low-stress puzzle box, with an open-ended nature which will have the tenacious gamer coming back over and over again, and give the competionist gamer a case of moon madness.

## Gameplay

The player comes into the game blind, seeing just the console. Right away they can glean a few things about their situation:

    1. They are drifting through space,

    2. The ship they are in is low tech,

    3. There is no ticking clock: no depleting oxygen or food supply
    
However, there is no obvious goal. The little story the game has is found by pushing a _lot_ of buttons.

There's no set way to "win" the game, but the player can lose, and lose spectacularly. 

## The Features

- **So Many Buttons:**
    The big draw of the game is to play with a lot of buttons. None of them are labelled, and it's not obvious what any of them should do. "Low key frustration" is the mood I'm going for.
        - Half of them will just make a funny noise (unfinished as of 10/2/22)
        - Each button has at least an *on* and *off* position.
    
    The coding goal is to have a few functions which work for every button, then some specialized events.

- **Animations:**
    There will be some animations for figuring out complicated puzzles, or losing. I'd also like to add ambient animation (example: flickering lights).

- **Note-taking:** A feature I'd like to add but not sure if I'll have time. Like I said, none of the buttons are labelled. But I'd like to give the players a way to take notes in-game. This is a stretch goal.

- **Languages**
    1. HTML
    2. CSS
    3. Javascript

- **Programs**
    1. VSCode
    2. Photoshop CC

## Bugs

9/27/22
Can't really zoom into game without the buttons going haywire

Buttons move around when crossing browsers, have to view in Chrome for best experience

10/1/22
Throttle isn't perfectly aligned.

10/2/22
- After finishing the keypad puzzle, you have to click the throttle a few times to warp.
- If you go to the planet background before hitting the self destruct, the old bg will be on the doom page.
- The frame scrolls with the browser while the game doesn't.
- Not a bug, but I still need to finish the buttons on the border

10/3/22
- Border isn't centered correctly.

## New Things I Learned
- math.random
- the includes keyword
- how to take elements out of an array dynamically
- some about "in" though I'm still not 100% on that
- adding audio

## Citations

-Special thanks to JD Howell, who helped with debugging.

-"Secret"'s comment on this conversation https://css-tricks.com/snippets/css/orientation-lock/ helped with turning orientation code.

-JS Web Game by _jlboba_ inspired the button functions, git repo here:
https://github.com/HackerUSA-CE/JS-Web-Game-Part-7.git

-W3Schools taught me about shake animation: https://www.w3schools.com/howto/howto_css_shake_image.asp

-Matthew Telles aided greatly in the writing of the checkSolution function

-AmerllicA's comment on this stack overflow conversation clarified taking an element from an array: https://stackoverflow.com/questions/3954438/how-to-remove-item-from-array-by-value

-All of the art for the game was made by me. More of my art can be found online under the iesnoth moniker. This is my first foray into pixel art.

## API

There is no API for this build.

## Licensing

TBD

CHANGELOG.md
# Changelog
## Single designer: Victoria Beasley

### March 10, 2023
Added a typescript branch to do TS upgrade. Will not be pushing to main branch, as it is too buggy at the moment

### March 8, 2023
Added a sass branch to do SCSS upgrade. Will not be pushing to main branch, as it is too buggy at the moment

## Oct 2, 2022

### General clean-up, removed comments and added clarifying comments, updated the readme

### index.html
Added last minute border jpg for the aesthetic, and for later functionality
    - info on game, reloading the game // these features don't work

### style.css
Added border

### CREATED doom.js
Added buttons to the border // they do not function

### doom.css
Added border

### index.js
Put buttons into arrays for readability and access
Added throttleFunction() to create/place throttle
    - figured it out after a lot of debugging, but have to click throttle twice for it to work

### puzzleFunctions.js
Functions (checkSolution, unlock) added for a puzzle in the game that unlocks the throttle

## Sept 30, 2022

### CREATED doom.html
Added page which holds the lose-page gif

### CREATED doom.css
Added css for basic styling

### index.js
Continued editing of button positions

### puzzleFunctions.js
Reworking the destructButton() functionality
Edited doom()

## Sept 29, 2022

### CREATED buttonCreation.js
Added (newButton, buttonFunction, move) functions for easier access
Edited buttonFunction() to accept a second id attribute
Created then deleted a function specific to switches: not DRY

### CREATED puzzleFunctions.js
Added functions (doom, destructButton) to randomly choose a button to trigger the doom page

### index.js
Added new id attribute to all buttons
Deleted button functions, since they are now in their own file

## Sept 28, 2022

### index.js
Change all instances of newButton() to buttonFunction(), which calls newButton inside it
    - buttonFunction() allows the buttons to be turned off and on

### style.css
Z-indexes and button div positions edited

## Sept 26, 2022

### index.js
Added all the needed buttons, spent a lot of time adjusting positions

### style.css
Added style for added button divs

## Sept 26, 2022

### index.js
Created (newButton, buttonFunction, move) functions
    - edited newButton
Added an object const for background assets
Begin positioning the plethora of buttons with (move, newButton)
    - 16 added

### index.html
Reuploaded the background assets in a different resolution

### style.css
Deleted (#circleDash) block
Edit the background css
Start positioning the panels of buttons through their divs

## Sept 25, 2022

### style.css
More styling for basic set up,
worked on sizing for button elements

### index.html
Put all components into divs for easier styling

## Sept 24, 2022

### index.html
Linked images

### style.css
Added basic styling

## Sept 21, 2022

### CREATED index.html
HTML boilerplate and pseudocode

### CREATED readme.md
Started writing readme

### CREATED script.js
Added pseudocode 

### CREATED style.css

script.js
test()
/*PSUEDO CODE

Basic list of what I'll need:

Functions:
flips a button from one position to another
plays a noise when a button is pressed
animates a lever when pressed

So, I want an event listener that changes the image of the button when the button is pressed.
The button is already created, and has a class of "switch" and value of "off"
I want to change the value to "on" in case they want to switch it back.
Some of the buttons look the same, so the same function could work for any of them.
HOWEVER, I need to make sure they don't all fire at the same time.

I could put all the alike buttons into their own arrays, and choose those for a forEach function, and have the two urls as arguments.
I would need a different function for the switches.

Function which appends a new image
function newButton(url){
    let button = document.createElement('button')
    image.src = url
    image.style.position = 'absolute'
    document.body.append(image)
    return image
}

Function that will give buttons functionality
function newItem(url){
    let item = newImage(url)
    item.addEventListener('click', () => {
        item.remove()
        let inventoryItem = document.createElement('img')
        inventoryItem.src = url;
        inventory.append(inventoryItem)
    })
    return item
}

Function that defines where a button will be
    function moveTo(left, bottom) {
        element.style.left = left + 'px'
        element.style.bottom = bottom + 'px'
    }


Puzzles:
A panel requires a certain pattern to be pressed in order for an event to occur

An event happens after a series of buttons is pressed in the right order


Stretch goals:
Add a border with different links as buttons
Put a credits and a way to handle volume on a different page

 */


buttonCreation.js

// makes the buttons and gives attributes
function newButton(url, id, id2) {
    let button = document.createElement('img')
    button.src = url
    button.style.position = 'relative';
    button.setAttribute('class', 'switch');
    button.setAttribute('value', 'off');
    button.setAttribute('id', id2);
    document.getElementById(id).appendChild(button);
    return button
}



//give buttons functionality
function buttonFunction(url1, url2, id, id2) {
    let button = newButton(url1, id, id2)
    //turns buttons off and on, adds function
    button.addEventListener('click', (e) => {
        if (e.target.getAttribute("value") == "off") {
            e.target.src = url2
            e.target.setAttribute('value', 'on')
            if (e.target.id >= 21 && e.target.id <= 29) {
                myVar.pressedKeys.push(e.target.id)
            }
            else if (e.target.id >= 9 && e.target.id <= 16) {
                myVar.abbaPressedKeys.push(e.target.id)
            }
        }
        else {
            e.target.src = url1
            e.target.setAttribute('value', 'off')
            if (e.target.id >= 21 && e.target.id <= 29) {
                myVar.pressedKeys.indexOf(e.target.id) !== -1 && myVar.pressedKeys.splice(myVar.pressedKeys.indexOf(e.target.id), 1)
            }
            else if (e.target.id >= 9 && e.target.id <= 16) {
                myVar.abbaPressedKeys.indexOf(e.target.id) !== -1 && myVar.abbaPressedKeys.splice(myVar.abbaPressedKeys.indexOf(e.target.id), 1)
            }
        }
        checkSolution()
        checkSolutionAbba()
    })
    return button
}



//specifically for the throttle

function throttleFunction(url1, url2, id, id2) {
    let button = newButton(url1, id, id2)
    //might need to delete this
    button.setAttribute('class', 'throttle')
    //will turn buttons off and on, add function
    button.addEventListener('click', (e) => {
        if (e.target.getAttribute("value") == "on") {
            e.target.setAttribute('value', 'warp')
            e.target.src = url2
        }
        else if (e.target.getAttribute("value") == 'warp') {
            warpSpeed()
            e.target.setAttribute('value', 'locked')
        }
        else if (e.target.getAttribute("value") == "locked") {
            window.alert(`We're out of fuel, genius.`)
        }
    })
    return button
}

//for the border buttons
//I'll figure this out later
// function borderButtons(id, url1, url2) {
//     let buttonB = document.getElementById(id)
//     buttonB.addEventListener('mouseover', (e) => {
//         e.target.src = url1
//     })
//     buttonB.addEventListener('mouseoout', (e) => {
//         e.target.src = url2
//     })
//     buttonB.addEventListener('click',(e) => {
//         console.log('clicked')
//     })
// }
// borderButtons('restart', 'assets/border-buttons/restart-dark.png', 'assets/border-buttons/restart.png')


//defines where a button will be
function move(element) {

    function moveTo(left, bottom) {
        element.style.left = left + 'px'
        element.style.bottom = bottom + 'px'
    }

    return {
        to: moveTo
    }
}

puzzleFunctions.js

const myVar = {
    pressedKeys: [],
    keypad: document.querySelector('#keypad'),
    keys: this.keypad.children,
    padSolution: ['21', '23', '25', '27', '29'],
    abbaPressedKeys: [],
    abbaSolution: ['9', '11', '13', '15'],
    audio:new Audio('assets/abba.mp3')

}

//function chooses a random button
function doom() {
    myVar.doomNumber = Math.floor(Math.random() * 47) + 1;
    //turn this off when played by others
    console.log(myVar.doomNumber)
    return myVar.doomLink
}

//check padSolution array against pressedKeys array
function checkSolution() {
    solved = true;
    for (var i = 0; i < myVar.padSolution.length; i++) {
        value = myVar.padSolution[i]
        if (!myVar.pressedKeys.includes(value)) { solved = false }
    }
    for (var i = 0; i < myVar.pressedKeys.length; i++) {
        value = myVar.pressedKeys[i]
        if (!myVar.padSolution.includes(value)) { solved = false }
    }
    if (solved) {
        window.alert('Throttle unlocked!')
        unlock()
    }
    else {
        console.log('not solved');
    }

}

//unlocks the throttle
function unlock() {
    document.getElementById('throttle-left').setAttribute('value','on')
    document.getElementById('throttle-right').setAttribute('value','on')
}

//uses changebg function to go to warp speed when throttle is thrown
function warpSpeed() {
        background = document.getElementById('bg');
        if(background.src = 'assets/sky.gif')
        {changeBg('assets/warp.gif');
        setTimeout(() =>{
            changeBg('assets/planet-gif.gif')
        },4000)
    }
        else if (background.src = 'assets/planet-gif.gif')
        {changeBg('assets/warp.gif');
        setTimeout(() =>{
            changeBg('assets/sky.gif')
        },4000)     
        }
}

//changes the background
function changeBg(url) {
    background = document.getElementById('bg')
    background.src = url
}

//sends user to the doom window
function destructButton() {
    doom()
    let element = document.getElementById(myVar.doomNumber);
    element.addEventListener('click', () => {
        window.location.href = 'doom.html'

    })
}
destructButton()


//***********abba puzzle */
//check abbaSolution against abbaPressed Keys
//plays 'dancing queen' when abbaPressedKeys = abbaSolution

function checkSolutionAbba() {
    solvedAbba = true;
    for (var i = 0; i < myVar.abbaSolution.length; i++) {
        value = myVar.abbaSolution[i]
        if (!myVar.abbaPressedKeys.includes(value)) { solvedAbba = false }
    }
    for (var i = 0; i < myVar.abbaPressedKeys.length; i++) {
        value = myVar.abbaPressedKeys[i]
        if (!myVar.abbaSolution.includes(value)) { solvedAbba = false }
    }
    if (solvedAbba) {
        myVar.audio.play();
    }
    else {
        myVar.audio.pause();
        console.log('not solved');
    }

}

src/buttonCreation.ts

// makes the buttons and gives attributes
//imputs are all strings
//returns an HTML object

function newButton(url: string, id: string, id2: string): HTMLImageElement {
    let button: HTMLImageElement = document.createElement('img')
    button.src = url
    button.style.position = 'relative';
    button.setAttribute('class', 'switch');
    button.setAttribute('value', 'off');
    button.setAttribute('id', id2);
    // the ! forces the TS to ignore the possibility of null and move on
    document.getElementById(id)!.appendChild(button);
    return button
}



//give buttons functionality
function buttonFunction(url1: string, url2: string, id: string, id2: string): HTMLImageElement {
    let button: HTMLImageElement = newButton(url1, id, id2)
    //turns buttons off and on, adds function
    button.addEventListener('click', (e: MouseEvent) => {
        if (e.target instanceof Element) {
            if (e.target.getAttribute("value") == "off") {
                e.target.src = url2
                e.target.setAttribute('value', 'on')
                if (typeof e.target.id == 'number') {
                    if (e.target.id >= 21 && e.target.id <= 29) {
                        myVar.pressedKeys.push(e.target.id)
                    }
                    else if (e.target.id >= 9 && e.target.id <= 16) {
                        myVar.abbaPressedKeys.push(e.target.id)
                    }
                }
            }
        }
        else {
            if (e.target instanceof Element) {
                e.target.src = url1
                e.target.setAttribute('value', 'off')
                if (typeof e.target.id == 'number') {
                    if (e.target.id >= 21 && e.target.id <= 29) {
                        myVar.pressedKeys.indexOf(e.target.id) !== -1 && myVar.pressedKeys.splice(myVar.pressedKeys.indexOf(e.target.id), 1)
                    }
                    else if (e.target.id >= 9 && e.target.id <= 16) {
                        myVar.abbaPressedKeys.indexOf(e.target.id) !== -1 && myVar.abbaPressedKeys.splice(myVar.abbaPressedKeys.indexOf(e.target.id), 1)
                    }
                }
            }
        }
        checkSolution()
        checkSolutionAbba()
    })
    return button
}



//specifically for the throttle

function throttleFunction(url1: string, url2: string, id: string, id2: string): unknown {
    let button = newButton(url1, id, id2)
    //might need to delete this
    button.setAttribute('class', 'throttle')
    //will turn buttons off and on, add function
    button.addEventListener('click', (e: MouseEvent) => {
        if (e.target instanceof Element) {
            if (e.target.getAttribute("value") == "on") {
                e.target.setAttribute('value', 'warp')
                e.target.src = url2
            }
            else if (e.target.getAttribute("value") == 'warp') {
                warpSpeed()
                e.target.setAttribute('value', 'locked')
            }
            else if (e.target.getAttribute("value") == "locked") {
                window.alert(`We're out of fuel, genius.`)
            }
        }
    })
    return button
}

//for the border buttons
//I'll figure this out later
// function borderButtons(id, url1, url2) {
//     let buttonB = document.getElementById(id)
//     buttonB.addEventListener('mouseover', (e) => {
//         e.target.src = url1
//     })
//     buttonB.addEventListener('mouseoout', (e) => {
//         e.target.src = url2
//     })
//     buttonB.addEventListener('click',(e) => {
//         console.log('clicked')
//     })
// }
// borderButtons('restart', 'assets/border-buttons/restart-dark.png', 'assets/border-buttons/restart.png')


//defines where a button will be
//element is another function which also calls another function
function move(element): object {
    //left and bottom are both numbers
    function moveTo(left: number, bottom: number) {
        element.style.left = left + 'px'
        element.style.bottom = bottom + 'px'
    }

    return {
        to: moveTo
    }
}

src/puzzleFunctions.ts

const myVar = {
    pressedKeys: [],
    keypad: document.querySelector('#keypad'),
    keys: this.keypad.children,
    padSolution: ['21', '23', '25', '27', '29'],
    abbaPressedKeys: [],
    abbaSolution: ['9', '11', '13', '15'],
    audio:new Audio('assets/abba.mp3')

}

//function chooses a random button
function doom():number {
    myVar.doomNumber = Math.floor(Math.random() * 47) + 1;
    //turn this off when played by others
    console.log(myVar.doomNumber)
    return myVar.doomLink
}

//check padSolution array against pressedKeys array
function checkSolution() {
    solved = true;
    for (var i = 0; i < myVar.padSolution.length; i++) {
        value = myVar.padSolution[i]
        if (!myVar.pressedKeys.includes(value)) { solved = false }
    }
    for (var i = 0; i < myVar.pressedKeys.length; i++) {
        value = myVar.pressedKeys[i]
        if (!myVar.padSolution.includes(value)) { solved = false }
    }
    if (solved) {
        unlock()
    }
    else {
        console.log('not solved');
    }

}

//unlocks the throttle
function unlock() {
    document.getElementById('throttle-left').setAttribute('value','on')
    document.getElementById('throttle-right').setAttribute('value','on')
}

//uses changebg function to go to warp speed when throttle is thrown
function warpSpeed() {
        background = document.getElementById('bg');
        if(background.src = 'assets/sky.gif')
        {changeBg('assets/warp.gif');
        setTimeout(() =>{
            changeBg('assets/planet-gif.gif')
        },4000)
    }
        else if (background.src = 'assets/planet-gif.gif')
        {changeBg('assets/warp.gif');
        setTimeout(() =>{
            changeBg('assets/sky.gif')
        },4000)     
        }
}

//changes the background
function changeBg(url) {
    background = document.getElementById('bg')
    background.src = url
}

//sends user to the doom window
function destructButton() {
    doom()
    let element = document.getElementById(myVar.doomNumber);
    element.addEventListener('click', () => {
        window.location.href = 'doom.html'

    })
}
destructButton()


//***********abba puzzle */
//check abbaSolution against abbaPressed Keys
//plays 'dancing queen' when abbaPressedKeys = abbaSolution

function checkSolutionAbba() {
    solvedAbba = true;
    for (var i = 0; i < myVar.abbaSolution.length; i++) {
        value = myVar.abbaSolution[i]
        if (!myVar.abbaPressedKeys.includes(value)) { solvedAbba = false }
    }
    for (var i = 0; i < myVar.abbaPressedKeys.length; i++) {
        value = myVar.abbaPressedKeys[i]
        if (!myVar.abbaSolution.includes(value)) { solvedAbba = false }
    }
    if (solvedAbba) {
        myVar.audio.play();
    }
    else {
        myVar.audio.pause();
        console.log('not solved');
    }

}

index.js
const switchArray = [
    [
        //Bottom left sliders
        move(buttonFunction('assets/slider/slider-down.png', 'assets/slider/slider-up.png', 'center-left',41)).to(0, -63),
        move(buttonFunction('assets/slider/slider-down.png','assets/slider/slider-up.png', 'center-left',42)).to(-7, -60),
        move(buttonFunction('assets/slider/slider-down.png','assets/slider/slider-up.png', 'center-left',43)).to(-6, -58),
        move(buttonFunction('assets/slider/slider-down.png','assets/slider/slider-up.png', 'center-left',44)).to(-5, -54),
        move(buttonFunction('assets/slider/slider-down.png', 'assets/slider/slider-up.png', 'center-left', 45)).to(-5, -52),
        //Center throttle
        move(throttleFunction('assets/throttle/left.png', 'assets/throttle/left-up.png', 'center-console','throttle-left')).to(18, -122),
        move(throttleFunction('assets/throttle/right.png','assets/throttle/right-up.png','center-console','throttle-right')).to(21, -120),
        //center dials
        move(buttonFunction('assets/center-dial/dial.png', 'assets/center-dial/turn.png', 'center-console',46)).to(-146, -55),
        move(buttonFunction('assets/center-dial/dial.png', 'assets/center-dial/turn.png', 'center-console',47)).to(0, -55),

    ]

]

const buttonArray = [
    //border buttons(removing until they work right)
    // [
    //     move(newButton('assets/border-buttons/restart.png', 'border-div', 'restart')).to(386, 73),
    //     move(newButton('assets/border-buttons/info.png', 'border-div', 'info')).to(395, 73)
    // ]
    //Top Console
    [
        //Left panel
        //Top Row
        move(buttonFunction('assets/switch-tl/off.png', 'assets/switch-tl/on.png', 'left-buttons', 1)).to(19, -30),
        move(buttonFunction('assets/switch-tl/off.png', 'assets/switch-tl/on.png', 'left-buttons', 2)).to(23, -19),
        move(buttonFunction('assets/switch-tl/off.png', 'assets/switch-tl/on.png', 'left-buttons', 3)).to(29, -8),
        //Bottom Row
        move(buttonFunction('assets/switch-tl/off.png', 'assets/switch-tl/on.png', 'left-buttons', 4)).to(-44, -74),
        move(buttonFunction('assets/switch-tl/off.png', 'assets/switch-tl/on.png', 'left-buttons', 5)).to(-40, -64),
        move(buttonFunction('assets/switch-tl/off.png', 'assets/switch-tl/on.png', 'left-buttons', 6)).to(39, -6)
    ],
    [
        //Circle TV
        move(buttonFunction('assets/circle-tv/off.png', 'assets/circle-tv/on.png', 'top', 7)).to(-124, 287),
        move(buttonFunction('assets/circle-tv/off.png', 'assets/circle-tv/on.png', 'top', 8)).to(-62, 289)
    ],
    //Center Panel
    //top row
    [
        move(buttonFunction('assets/center-switch/off.png', 'assets/center-switch/on.png', 'center-buttons', 9)).to(24, -7),
        move(buttonFunction('assets/center-switch/off.png', 'assets/center-switch/on.png', 'center-buttons', 10)).to(30, -7),
        move(buttonFunction('assets/center-switch/off.png', 'assets/center-switch/on.png', 'center-buttons', 11)).to(37, -7),
        move(buttonFunction('assets/center-switch/off.png', 'assets/center-switch/on.png', 'center-buttons', 12)).to(44, -7),
        //bottom row
        move(buttonFunction('assets/center-switch/off.png', 'assets/center-switch/on.png', 'center-buttons', 13)).to(-58, -54),
        move(buttonFunction('assets/center-switch/off.png', 'assets/center-switch/on.png', 'center-buttons', 14)).to(-52, -54),
        move(buttonFunction('assets/center-switch/off.png', 'assets/center-switch/on.png', 'center-buttons', 15)).to(-45, -54),
        move(buttonFunction('assets/center-switch/off.png', 'assets/center-switch/on.png', 'center-buttons', 16)).to(40, -18)
    ],
    [    //Right panel
        move(buttonFunction('assets/circle-tr/off.png', 'assets/circle-tr/on.png', 'right-buttons', 17)).to(19, -18),
        move(buttonFunction('assets/circle-tr/off.png', 'assets/circle-tr/on.png', 'right-buttons', 18)).to(30, -29),
        move(buttonFunction('assets/circle-tr/off.png', 'assets/circle-tr/on.png', 'right-buttons', 19)).to(-33, -49),
        move(buttonFunction('assets/circle-tr/off.png', 'assets/circle-tr/on.png', 'right-buttons', 20)).to(-23, -60)
    ],
    //Bottom Panel
    [
        //KEYPAD
        //top row
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 21)).to(-3, -5),
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 22)).to(-8, -5),
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 23)).to(-13, -5),
        //middle row
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 24)).to(5, 4),
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 25)).to(0, 4),
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 26)).to(-5, 4),
        //bottom row
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 27)).to(13, 12),
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 28)).to(7, 12),
        move(buttonFunction('assets/keypad-left/up.png', 'assets/keypad-left/down.png', 'keypad', 29)).to(2, 12)
    ],
    [
        //Big Squares
        move(buttonFunction('assets/big-square/up.png', 'assets/big-square/down.png', 'big-keys', 30)).to(-21, -7),
        move(buttonFunction('assets/big-square/up.png', 'assets/big-square/down.png', 'big-keys', 31)).to(15, 40),
        move(buttonFunction('assets/big-square/up.png', 'assets/big-square/down.png', 'big-keys', 32)).to(-10, 49),
        move(buttonFunction('assets/big-square/up.png', 'assets/big-square/down.png', 'big-keys', 33)).to(26, 95)
    ],
    //RIGHT SIDE
    //toggles
    [
        move(buttonFunction('assets/toggle-r/up.png', 'assets/toggle-r/down.png', 'center-right', 34)).to(80, -40),
        move(buttonFunction('assets/toggle-r/up.png', 'assets/toggle-r/down.png', 'center-right', 35)).to(90, -40),
        move(buttonFunction('assets/toggle-r/up.png', 'assets/toggle-r/down.png', 'center-right', 36)).to(97, -40)
    ],
    //tiny lights
    [
        move(buttonFunction('assets/tiny-r/off.png', 'assets/tiny-r/on.png', 'center-right', 37)).to(-51, -73),
        move(buttonFunction('assets/tiny-r/off.png', 'assets/tiny-r/on.png', 'center-right', 38)).to(-20, -75)
    ],
    //knobs with lights
    [
        move(buttonFunction('assets/knob/off.png', 'assets/knob/on.png', 'center-right', 39)).to(-2, -78),
        move(buttonFunction('assets/knob/off.png', 'assets/knob/on.png', 'center-right', 40)).to(2, -80)
    ]

]

//border buttons
move(buttonFunction('assets/border-buttons/restart.png', 'assets/border-buttons/restart-dark.png', 'border-div', 'restart')).to(380, 73)
move(buttonFunction('assets/border-buttons/info.png', 'assets/border-buttons/infodark.png', 'border-div', 'info')).to(382, 73)


